snap/validate.go

```diff
@@ -781,3 +786,16 @@ func ValidateLayout(layout *Layout, constraints []LayoutConstraint) error {
 	}
 	return nil
 }
+
+func ValidateCommonIDs(info *Info) error {
+	seen := make(map[string]bool, len(info.Apps))
+	for _, app := range info.Apps {
+		if app.CommonID != "" {
+			if seen[app.CommonID] {
+				return fmt.Errorf("application %q common-id %q is not unique", app.Name, app.CommonID)
```

Comment:
"application %q common-id %q must be unique"

plus possibly

", already used by application %q"

Reply:
Yes, that's a nice touch.